GitCommands/Git/GitModule.cs

```diff
@@ -3360,14 +3280,19 @@ public IEnumerable<string> GetPreviousCommitMessages(string revision, int count)
                 });
         }
 
-        public string OpenWithDifftool(string filename, string oldFileName = "", string firstRevision = GitRevision.IndexGuid, string secondRevision = GitRevision.UnstagedGuid, string extraDiffArguments = "", bool isTracked = true)
+        public string OpenWithDifftool(string filename, string oldFileName = "", string firstRevision = GitRevision.IndexGuid, string secondRevision = GitRevision.UnstagedGuid, string extraDiffArguments = null, bool isTracked = true)
         {
-            const string output = "";
+            var args = new ArgumentBuilder
+            {
+                "difftool --gui --no-prompt",
+                extraDiffArguments,
+                _revisionDiffProvider.Get(firstRevision, secondRevision, filename, oldFileName, isTracked)
+            };
 
-            string args = string.Join(" ", extraDiffArguments, _revisionDiffProvider.Get(firstRevision, secondRevision, filename, oldFileName, isTracked));
-            RunGitCmdDetached("difftool --gui --no-prompt " + args);
+            RunGitCmdDetached(args.ToString());
 
-            return output;
+            // TODO why bother returning this?
```

Comment:
This is supposed to return any possible error message from the difftool
Maybe we can remove the return

Reply:
I see. `RunGitCmdDetached` doesn't wait for any process output anyway, so yes it seems redundant. Open questions exist about whether something should track the returned `Process` object, but that feels like a different PR.

I am doing a bunch of work towards the goal of having a single point through which all git executions occur. I can cover this under that chunk of work.

For now I've extended the TODO comment.